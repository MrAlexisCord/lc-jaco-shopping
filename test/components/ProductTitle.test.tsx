import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { ProductTitle } from '../../src/components/ProductTitle';
import { product1 } from '../data/products';


describe('TESTs on ProductTitle', () => {
    test('***   Debe mostrar el component con el título personalizado', () => {
        const wrapper = renderer.create(
            <ProductTitle title='Custom Product' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('***   Debe mostrar el component con el título del producto (from context)', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>{
                () => (<ProductTitle />)
            }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
})