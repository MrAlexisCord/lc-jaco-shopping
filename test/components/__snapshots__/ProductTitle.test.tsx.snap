// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTs on ProductTitle ***   Debe mostrar el component con el título del producto (from context) 1`] = `
<div
  className="productCard "
>
  <span
    className="productDescription "
  >
    Coffee Mug
  </span>
</div>
`;

exports[`TESTs on ProductTitle ***   Debe mostrar el component con el título personalizado 1`] = `
<span
  className="productDescription "
>
  Custom Product
</span>
`;
